x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -31,7 +31,7 @@ import {
 import { AuthenticationResult } from './authentication_result';
 import { DeauthenticationResult } from './deauthentication_result';
 import { Tokens } from './tokens';
-import { SessionInfo } from '../../public/types';
+import { SessionInfo } from '../../public';
```

Comment:
FWIW, I like the changes you made. Did the lint rule require these changes?

Reply:
Yes it does. If you don't want this to be part of the public contract, we can move this up into a shared `types` file that only security can import from.

Reply:
BTW shouldn't we ban public code to be imported from server as well? I'm concern about cases when a plugin accesses browser specific API and can throw on the server side

Reply:
I'm not against that, was just focusing this change on fixing the Webpack bundling problems. It looks like there are more violations of this pattern, let me see about doing that change in a separate PR.